http_client/models/http_client_auth.py

```diff
@@ -120,85 +122,89 @@ class HttpClientAuth(models.AbstractModel):
         "Token Verify Path",
         help="If set, will be able to verify existing access token validity",
     )
-    company_id = fields.Many2one('res.company', default=lambda s: s.env.user.company_id)
+    company_id = fields.Many2one("res.company", default=lambda s: s.env.user.company_id)
     state = fields.Selection(
-        [('draft', "Not Confirmed"), ('confirmed', "Confirmed")],
-        default='draft',
+        [("draft", "Not Confirmed"), ("confirmed", "Confirmed")],
+        default="draft",
         copy=False,
         readonly=True,
         required=True,
     )
+    api_key_header = fields.Char(
+        "API Key header",
```

Comment:
We don't really need new field here. We can use `identifier`. Api Key and Basic auth both have key and secret. Its just the difference how they are used (and labels have little meaning behind the scenes).

Of course we can have onchange to auto fill `identifier` with API Key Header default to make it more convenient.